new.go

```diff
@@ -28,7 +29,20 @@ func newBuilder(store storage.Store, options BuilderOptions) (*Builder, error) {
 		name = options.Container
 	} else {
 		if image != "" {
-			name = image + "-" + name
+			prefix := image
+			s := strings.Split(prefix, "/")
+			if len(s) > 0 {
+				prefix = s[len(s)-1]
+			}
+			s = strings.Split(prefix, ":")
+			if len(s) > 0 {
+				prefix = s[0]
+			}
+			s = strings.Split(prefix, "@")
+			if len(s) > 0 {
+				prefix = s[0]
+			}
+			name = prefix + "-" + name
 		}
 	}
 	if name != "" {
```

Comment:
If we end up not passing in a name, the container gets created anyway, but can only be referred to by its ID.  This bit of logic is mainly about providing a friendly default if one isn't specified.